xptifw/src/xpti_trace_framework.cpp

```diff
@@ -806,9 +833,6 @@ class Framework {
     if (!Payload || !InstanceNo)
       return nullptr;
 
-    if (Payload->flags == 0)
```

Comment:
this code allows to have empty payload. That means that query payload by universal id could return structure object without any meaningful data. Is it intended change?
I found that this lead to the failures in Tracing/usm tests because previously check if Payload != nullptr was enough. 
To fix tests you should add to https://github.com/intel/llvm/blob/sycl/sycl/tools/xpti_helpers/usm_analyzer.hpp#L240
check that payload has needed data like
 ```
if (Payload && (Payload->flags &
         static_cast<uint64_t>(xpti::payload_flag_t::SourceFileAvailable))) {
```